engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsAnyNode.java

```diff
@@ -0,0 +1,574 @@
+      "otherInterop.isBoolean(otherBoolean)"
+  }, limit = "3")
+  boolean equalsBooleanInterop(
+      Object selfBoolean,
+      Object otherBoolean,
+      @CachedLibrary("selfBoolean") InteropLibrary selfInterop,
+      @CachedLibrary("otherBoolean") InteropLibrary otherInterop
+  ) {
+    try {
+      return selfInterop.asBoolean(selfBoolean) == otherInterop.asBoolean(otherBoolean);
+    } catch (UnsupportedMessageException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  @Specialization(guards = {
+      "selfInterop.isMetaObject(selfMeta)",
+      "otherInterop.isMetaObject(otherMeta)",
+  }, limit = "3")
+  boolean equalsMetaObjects(Object selfMeta, Object otherMeta,
+      @CachedLibrary("selfMeta") InteropLibrary selfInterop,
+      @CachedLibrary("otherMeta") InteropLibrary otherInterop,
+      @Cached EqualsAnyNode equalsNode) {
+    try {
+      Object selfMetaName = selfInterop.getMetaQualifiedName(selfMeta);
+      Object otherMetaName = otherInterop.getMetaQualifiedName(otherMeta);
+      return equalsNode.execute(selfMetaName, otherMetaName);
+    } catch (UnsupportedMessageException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  @Specialization(guards = {
+      "!selfInterop.isDate(selfTimeZone)",
+      "!selfInterop.isTime(selfTimeZone)",
+      "selfInterop.isTimeZone(selfTimeZone)",
+      "!otherInterop.isDate(otherTimeZone)",
+      "!otherInterop.isTime(otherTimeZone)",
+      "otherInterop.isTimeZone(otherTimeZone)"
+  }, limit = "3")
+  boolean equalsTimeZones(Object selfTimeZone, Object otherTimeZone,
+      @CachedLibrary("selfTimeZone") InteropLibrary selfInterop,
+      @CachedLibrary("otherTimeZone") InteropLibrary otherInterop) {
+    try {
+      return selfInterop.asTimeZone(selfTimeZone).equals(
+          otherInterop.asTimeZone(otherTimeZone)
+      );
+    } catch (UnsupportedMessageException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  @TruffleBoundary
+  @Specialization(guards = {
+      "selfInterop.isDate(selfZonedDateTime)",
+      "selfInterop.isTime(selfZonedDateTime)",
+      "selfInterop.isTimeZone(selfZonedDateTime)",
+      "otherInterop.isDate(otherZonedDateTime)",
+      "otherInterop.isTime(otherZonedDateTime)",
+      "otherInterop.isTimeZone(otherZonedDateTime)"
+  }, limit = "3")
+  boolean equalsZonedDateTimes(Object selfZonedDateTime, Object otherZonedDateTime,
+      @CachedLibrary("selfZonedDateTime") InteropLibrary selfInterop,
+      @CachedLibrary("otherZonedDateTime") InteropLibrary otherInterop) {
+    try {
+      return ZonedDateTime.of(
+          selfInterop.asDate(selfZonedDateTime),
+          selfInterop.asTime(selfZonedDateTime),
+          selfInterop.asTimeZone(selfZonedDateTime)
+      ).isEqual(
+          ZonedDateTime.of(
+              otherInterop.asDate(otherZonedDateTime),
+              otherInterop.asTime(otherZonedDateTime),
+              otherInterop.asTimeZone(otherZonedDateTime)
+          )
+      );
```

Comment:
Minor stylistic nitpick:
```suggestion
      var self = ZonedDateTime.of(
          selfInterop.asDate(selfZonedDateTime),
          selfInterop.asTime(selfZonedDateTime),
          selfInterop.asTimeZone(selfZonedDateTime)
      );
      var other = ZonedDateTime.of(
          otherInterop.asDate(otherZonedDateTime),
          otherInterop.asTime(otherZonedDateTime),
          otherInterop.asTimeZone(otherZonedDateTime)
      )
      return self.isEqual(other);
```
is IMO much more readable and should be more debugger-friendly as now you have given names to the intermediate values so it is easier to preview them when stepping through the program execution.

Reply:
I kind of agree. But I would rather keep this consistent with other specializations, that follow the similar pattern. Besides, `ZonedDateTime.isEquals` is a part of java std lib, so one will rarely need to debug it.

Reply:
Why are they using this pattern? My point was to change this in all specializations, as it would be more readable, I just used this one as example. Of course we should be consistent.

Reply:
> Why are they using this pattern?

Because I like it more - it is just one expression. It seems that you insist on this change. So let me integrate it. I guess that in the end, it is more useful.

Reply:
Ah! Sorry then, I misread your previous comment:

> I kind of agree

I thought that you agree that this is more readable. If you think the current approach is better I'm not strongly against keeping it.